Src/DependencyCollector/Shared/ComponentCorrelation.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Component Correlation configuration.
+    /// </summary>
+    public class ComponentCorrelation
```

Comment:
The class is public because the property that matches the config is public. 
Also it has to be it's own class rather than a property because of the way we decided to form the config: 
```<Add Type="Microsoft.ApplicationInsights.DependencyCollector.DependencyTrackingTelemetryModule, Microsoft.AI.DependencyCollector">
      <ComponentCorrelation>
        <CollectionEnabled>true</CollectionEnabled>
        <ExcludedDomains>
          <Add>www.bbc.com</Add>
          <Add>http://www.msn.com.cn/news</Add>
        </ExcludedDomains>
      </ComponentCorrelation>
    </Add>
```
Since the logic that processes the config is out of our control - in order to parse this, we have to define ComponentCorrelation as a property which in turn contains CollectionEnabled and ExcludedDomains.

Are you saying you would rather have something like the following - which would allow these to become properties on the modules
```
    <Add Type="Microsoft.ApplicationInsights.DependencyCollector.DependencyTrackingTelemetryModule, Microsoft.AI.DependencyCollector">
      <ComponentCorrelationCollectionEnabled>true</ComponentCorrelationCollectionEnabled>
      <ComponentCorrelationExcludedDomains>
        <Add>www.bbc.com</Add>
        <Add>http://www.msn.com.cn/news</Add>
      </ComponentCorrelationExcludedDomains>
    </Add>
```